torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -92,6 +92,18 @@ c10::optional<at::Device> pickDeviceType(
 } // namespace jit
 } // namespace torch
 
+size_t normalizeAndCheckIndex(int64_t idx, int64_t list_size) {
```

Comment:
I wonder if we should actually just use int64_t here.  I kinda forget why we're using size_t and the explicit-width types feel much more reliable.

Reply:
errr, okay, I tried, but softmax has too many comparisons with a size_t loop variable :-) This looks a bit less ugly, tbh